packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -13,81 +13,193 @@
+  }
+  return null;
+}
+
+NoProductsFound.propTypes = {
+  searchTerm: PropTypes.string,
+};
 
 function ShoppingPane(props) {
-  const isEnabled = useFeature('shoppingIntegration');
-  const [product, setProduct] = useState(null);
+  const isShoppingIntegrationEnabled = useFeature('shoppingIntegration');
+  const speak = useLiveRegion('assertive');
+  const [searchTerm, setSearchTerm] = useState('');
+  const [isMenuFocused, setIsMenuFocused] = useState(false);
+  const [onPageProducts, setOnPageProducts] = useState([]);
+  const [products, setProducts] = useState([]);
+  const {
+    actions: { getProducts },
+  } = useAPI();
+
+  const { currentPage } = useStory(({ state: { currentPage } }) => ({
+    currentPage,
+  }));
+
+  useEffect(() => {
+    setOnPageProducts(getOnPageProducts(currentPage));
+  }, [currentPage]);
+
+  const getProductsByQuery = useCallback(
+    async (value = '') => {
+      try {
+        setProducts(await getProducts(value));
+      } catch (err) {
+        setProducts([]);
+      }
+    },
+    [getProducts]
+  );
+
+  const onSearch = useCallback(
+    (evt) => {
+      const value = evt.target.value;
+      if (value !== searchTerm) {
+        setSearchTerm(value);
+        getProductsByQuery(value);
+      }
+    },
+    [searchTerm, setSearchTerm, getProductsByQuery]
+  );
+
+  useEffect(() => {
+    getProductsByQuery();
+  }, [getProductsByQuery]);
+
+  const handleInputKeyPress = useCallback((event) => {
+    const { key } = event;
+    if (key === 'ArrowDown' || key === 'Tab') {
+      setIsMenuFocused(true);
+    }
+  }, []);
+
+  const handleFocus = () => {
+    setIsMenuFocused(false);
+  };
+
+  const { deleteElementById } = useStory(({ actions }) => ({
+    deleteElementById: actions.deleteElementById,
+  }));
 
   const { insertElement } = useLibrary((state) => ({
     insertElement: state.actions.insertElement,
   }));
 
-  const insertProduct = useCallback(() => {
-    insertElement('product', {
-      width: 25,
-      height: 25,
-      product,
-    });
-    setProduct(null);
-  }, [insertElement, product]);
+  const insertProduct = useCallback(
+    (product) => {
+      insertElement('product', {
+        width: 25,
+        height: 25,
+        product,
+      });
+
+      const PRODUCT_ADDED_TEXT = sprintf(
+        /* translators: %s: product title added to page.*/
+        __('%s added to page', 'web-stories'),
+        product.productTitle
+      );
+
+      speak(PRODUCT_ADDED_TEXT);
+    },
+    [insertElement, speak]
+  );
+
+  const deleteProduct = useCallback(
+    (product) => {
+      const element = onPageProducts.find(
+        (item) => item.productId === product.productId
+      );
+      if (element) {
+        deleteElementById({ elementId: element.elementId });
+      }
+    },
+    [deleteElementById, onPageProducts]
+  );
+
+  const handleClearInput = useCallback(() => {
+    setSearchTerm('');
+  }, [setSearchTerm]);
```

Comment:
We need to call `getProductsByQuery();` here too.

Otherwise, after I search for "ssgdsgdrgdrgaerdhgrdth" and then clear the input field, I am stuck at the "No products found" state.

Or actually, I don't even see the "No products found" message, so that's a bug:

![Screenshot 2022-05-10 at 14 47 26](https://user-images.githubusercontent.com/841956/167631942-5e39810d-103a-4fee-b15b-84af69b1f101.png)